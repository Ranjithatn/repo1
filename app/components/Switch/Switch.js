import React from "react";

const Switch = ({ id, disabled = false, checked, text, className, value, onChange, name }) => {
  return (
    <div className="field">
      <input
        id={id ? id : ""}
        type="checkbox"
        onChange={onChange}
        disabled={disabled}
        value={value}
        name={id ? id : ""}
        className={"switch" + (className ? " " + className : "")}
        checked={checked}
      />
     <label  htmlFor={id ? id : ""}>{text}</label>
    </div>
  );
};

export default Switch;
