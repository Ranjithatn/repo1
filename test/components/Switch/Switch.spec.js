import React from "react";
import toJson from "enzyme-to-json";
import Switch from "../../../app/components/Switch/Switch";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

describe('<Switch />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Switch  />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
