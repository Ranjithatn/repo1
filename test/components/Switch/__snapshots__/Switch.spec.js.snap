// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render without crashing 1`] = `
<Switch>
  <div
    className="field"
  >
    <input
      className="switch"
      disabled={false}
      id=""
      name=""
      type="checkbox"
    />
    <label
      htmlFor=""
    />
  </div>
</Switch>
`;
